src/index.ts

```diff
@@ -0,0 +1,350 @@
+            },
+            {
+                name: "domain",
+                alias: "d",
+                description:
+                    "{underline TopDomainChecker prioritizes arguments over standard inputs}. \n If it missing, show the interactive prompt. \n Also, {bold --domain} is specified as the default argument and is not required."
+            }
+        ]
+    },
+    {
+        content: `Project home: ${terminalLink(
+            "GitHub",
+            "https://github.com/P2P-Develop/TopDomainChecker"
+        )}`
+    }
+]);
+
+// Arguments cast interface
+type Arguments = {
+    version: boolean,
+    verbose: boolean,
+    domain: string[],
+    help: boolean,
+    quiet: boolean,
+    "no-box": boolean,
+    "add-tld": string[],
+    "dry-run": boolean
+}
+
+const arguments_ = commandLineArgs([
+    { name: "version", alias: "V", type: Boolean, defaultValue: false },
+    { name: "verbose", alias: "v", type: Boolean, defaultValue: false },
+    {
+        name: "domain",
+        alias: "d",
+        type: String,
+        multiple: true,
+        defaultOption: true,
+        defaultValue: [""]
+    },
+    { name: "help", alias: "h", type: Boolean, defaultValue: false },
+    { name: "quiet", alias: "q", type: Boolean, defaultValue: false },
+    { name: "no-box", alias: "Q", type: Boolean, defaultValue: false },
+    { name: "add-tld", alias: "t", type: String, defaultValue: [""], multiple: true },
+    { name: "dry-run", alias: "D", type: Boolean, defaultValue: false }
+]) as Arguments;
+
+if (arguments_.version) {
+    console.log("v3.1.1");
+    console.log(arguments_);
+    process.exit(0);
+}
+
+if (arguments_.help) {
+    // Show usages if has argument "--help"
+    console.log(usage);
+    process.exit(0);
+}
+
+if (!arguments_["dry-run"] && arguments_.quiet && arguments_.verbose) {
+    // quiet and verbose cannot be used at the same time without --dry-run
+    console.warn(`${chalk.yellowBright.inverse.bold(`  ${figures.warning}  `)} ${chalk.bold("--quiet")} and ${chalk.bold("--verbose")} cannot be used at same time! Replaced with default value!`);
+    arguments_.quiet = false;
+    arguments_.verbose = false;
+}
+
+const requestGet = () => {
+    let lockFlag = false;
+
+    const spinner = arguments_.verbose ? ora("Fetching top-level domains information from IANA...") : undefined;
+
+    if (spinner) {
+        spinner.start();
+    }
+
+    https.get("https://data.iana.org/TLD/tlds-alpha-by-domain.txt", (response) => {
+        response.on("data", (chunk) => {
+            if (!lockFlag) {
+                if (spinner) {
+                    spinner.succeed(`Fetching top-level domains information from IANA...${chalk.greenBright("Success")}`);
+                }
+
+                main([
+                    ...(`${chunk}`
+                        .toLowerCase()
+                        .split(/\r\n|\n|\r/)
+                        .filter(tld => !tld.startsWith("#") || !tld)),
+                    ...addTld
```

Comment:
Codacy found an issue: ['addTld' was used before it was defined.](https://app.codacy.com/gh/P2P-Develop/TopDomainChecker/pullRequest?prid=6364165)